packages/react-router/lib/hooks.tsx

```diff
@@ -832,7 +832,7 @@ export function _renderMatches(
         let { loaderData, errors } = dataRouterState;
         let needsToRunLoader =
           match.route.loader &&
-          loaderData[match.route.id] === undefined &&
+          !(match.route.id in loaderData) &&
```

Comment:
The presence of the key in `loaderData`/`actionData` now indicates if it ran or not